### PR TITLE
[GEOS-9862] : REST API - API generated with default name

### PR DESCRIPTION
**Pull request by eropartz**
Add a tag on each endpoint to have a dedicated name for the generated API
https://osgeo-org.atlassian.net/browse/GEOS-9862

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or not applicable.**

For all pull requests:

- [X] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [X] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [X] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [ ] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [ ] PR for bug fixes and small new features are presented as a single commit
- [ ] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [ ] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [ ] New unit tests have been added covering the changes
- [ ] This PR passes all existing unit tests (test results will be reported by Continuous Integration after opening this PR)
- [ ] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by Continuous Integration after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

**Review by aaime — COMMENTED** [review]
As far as I can tell, it's all about adding tags to the resources. Looks like a useful improvement. 
However, there is a mix of styles about the tags:
- Some are all lower case
- Some are camel case
- Some are using dashes to separate words

I would prefer to see a consistent camelcase approach, with an initial capital letter, e.g. "GwcGridsets".
Looking at OGC API - Features, the tags are using at least capital letters, thought I could not find examples of camel-case:
https://github.com/opengeospatial/ogcapi-features/blob/master/core/examples/openapi/ogcapi-features-1-example1.yaml

Thinking out loud, would it be possible to have a part of the build that tries to build a client? Not as a way to ship one mind, but more as a way to verify that a client can indeed still be generated.

**Comment by eropartz** [comment]
New commit with tags in camelcase. There is an example of generation configuration with the swagger-codegen-maven-plugin in the [PR](https://osgeo-org.atlassian.net/browse/GEOS-9862)

File: doc/en/api/1.0.0/opensearch-eo.yaml [hunk 1]
```diff
@@ -15,6 +15,8 @@ paths:
   /collections:
     get:
       description: Retrieves a list of all available collections (names and links)
+      tags:
+       - "OpenSearchEo"
```
Comment (aaime) [inline doc/en/api/1.0.0/opensearch-eo.yaml:19]: Should be OpenSearchEO. Generally, when a bit of the name is an acronym already, it's kept in capitals.

File: doc/en/api/1.0.0/owsservices.yaml [hunk 2]
```diff
@@ -17,6 +17,8 @@ paths:
 
     get:
       operationId: getWMSSettings
+      tags:
+       - "OwsServices"
```
Comment (aaime) [inline doc/en/api/1.0.0/owsservices.yaml:21]: "OWSServices"

File: doc/en/api/1.0.0/wmslayers.yaml [hunk 3]
```diff
@@ -16,6 +16,8 @@ paths:
 
     get:
       operationId: getWMSStoreLayers
+      tags:
+       - "WmsLayers"
```
Comment (aaime) [inline doc/en/api/1.0.0/wmslayers.yaml:20]: WMSLayers

File: doc/en/api/1.0.0/wmsstores.yaml [hunk 4]
```diff
@@ -17,6 +17,8 @@ paths:
 
     get:
       operationId: getWMSStores
+      tags:
+       - "WmsStores"
```
Comment (aaime) [inline doc/en/api/1.0.0/wmsstores.yaml:21]: WMSStores

File: doc/en/api/1.0.0/wmtslayers.yaml [hunk 5]
```diff
@@ -16,6 +16,8 @@ paths:
 
     get:
       operationId: getWMTSStoreLayers
+      tags:
+       - "WmtsLayers"
```
Comment (aaime) [inline doc/en/api/1.0.0/wmtslayers.yaml:20]: WMTSLayers.

File: doc/en/api/1.0.0/wmtsstores.yaml [hunk 6]
```diff
@@ -17,6 +17,8 @@ paths:
 
     get:
       operationId: getWMTSStores
+      tags:
+       - "WmtsStores"
```
Comment (aaime) [inline doc/en/api/1.0.0/wmtsstores.yaml:21]: WMTSStores

**Review by aaime — COMMENTED** [review]
Thanks, looks so much better! Found a few cases where acronyms have been lowercased, could you fix them too? Then the PR should be good for merging.

File: doc/en/api/1.0.0/owsservices.yaml (hunk 2 above)
Comment (eropartz) [reply doc/en/api/1.0.0/owsservices.yaml]: Done

File: doc/en/api/1.0.0/wmsstores.yaml (hunk 4 above)
Comment (eropartz) [reply doc/en/api/1.0.0/wmsstores.yaml]: Done

File: doc/en/api/1.0.0/wmtslayers.yaml (hunk 5 above)
Comment (eropartz) [reply doc/en/api/1.0.0/wmtslayers.yaml]: Done

File: doc/en/api/1.0.0/wmtsstores.yaml (hunk 6 above)
Comment (eropartz) [reply doc/en/api/1.0.0/wmtsstores.yaml]: Done

**Review by aaime — APPROVED** [review]
Looks good to me, merging.

**Pull request merged**